libs/rpc/protocols/object-protocol.js

```diff
@@ -0,0 +1,125 @@
+(function (window) {
+
```

Comment:
In order to pass message between content and background your object need to be easy for serialising/parsing by browser. First issue with serialising class instance is because it has methods and that is not allowed. Even if it serialises class instance on the other side it won't receive the same class instance, it will get basic object with properties. Receiver must know type of message it received to know how to handle it and to call its methods if needed.
We are not converting to string because it will be done by browser which is much faster.
Beside all of that this protocol can be used in many other types of communication in browser which have the same restrictions.